tests/drillhole_v4_0_test.py

```diff
@@ -732,3 +732,201 @@ def test_is_collocated(tmp_path):
     assert property_group.is_collocated(
         np.c_[np.arange(0, 10.0), np.arange(1, 11.0)], 0.01
     )
+
+
+def compare_structured_arrays(
+    array1: np.ndarray, array2: np.ndarray, tolerance: float = 1e-5
+) -> bool:
+    """
+    Compare two NumPy record arrays.
+
+    :param array1: The first structured array to compare.
+    :param array2: The second array to be compared.
+    :param tolerance: The tolerance level for numerical comparison.
+
+    :return: True if arrays are equivalent within the given tolerance, False otherwise.
+    """
+
+    if array1.dtype != array2.dtype:
+        return False  # Different structure
+
+    for column in array1.dtype.names:
+        data1, data2 = array1[column], array2[column]
+
+        # Decode byte strings to regular strings if necessary
+        if data1.dtype.kind == "S":
+            data1 = np.array([d.decode() for d in data1])
+            data2 = np.array([d.decode() for d in data2])
+
+        # Check if the data type of the column is numerical
+        if np.issubdtype(data1.dtype, np.number):
+            if not np.all(np.isclose(data1, data2, atol=tolerance, equal_nan=True)):
+                return False
+        else:
+            if not np.array_equal(data1, data2):
+                return False
+
+    return True
+
+
+def test_export_table(tmp_path):
+    h5file_path = tmp_path / r"test_drillholeGroup.geoh5"
+    well_name = "bullseye/"
+    n_data = 10
+
+    with Workspace.create(h5file_path) as workspace:
+        # Create a workspace
+        dh_group = DrillholeGroup.create(workspace)
+
+        well = Drillhole.create(
+            workspace,
+            collar=np.r_[0.0, 10.0, 10],
+            surveys=np.c_[
+                np.linspace(0, 100, n_data),
+                np.ones(n_data) * 45.0,
+                np.linspace(-89, -75, n_data),
+            ],
+            parent=dh_group,
+            name=well_name,
+        )
+
+        depth = np.sort(np.random.uniform(low=0.05, high=100, size=(10,))).astype(
+            np.float16
+        )
+        value = np.random.randn(8)
+
+        well.add_data(
+            {
+                "Depth Data": {
+                    "depth": depth,
+                    "values": value,
+                },
+            }
+        )

```

Comment:
In order to cover the case where some data are not present on certain drillhole, can you add another drillhole and another data to your test? Only one of the data should be shared by both drillholes.